<a href="https://colab.research.google.com/github/chaitu983/Datasets/blob/master/DPhi_Final_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas 
pandas.__version__



'1.1.5'

# TASK -**1**

In [ ]:
from google.colab import files
data_to_load = files.upload() 
import pandas as pd

import io

data_18 = pd.read_csv(io.BytesIO(data_to_load['Survey_2018.txt']))
print(data_18)


Saving Survey_2018.txt to Survey_2018 (6).txt
               Timestamp   Age  ... Company size   Company type
0    14/12/2018 12:41:33  43.0  ...     100-1000        Product
1    14/12/2018 12:42:09  33.0  ...       50-100        Product
2    14/12/2018 12:47:36  32.0  ...        1000+        Product
3    14/12/2018 12:50:15  25.0  ...        1000+        Product
4    14/12/2018 12:50:31  39.0  ...     100-1000  Ecom retailer
..                   ...   ...  ...          ...            ...
760  03/06/2020 20:12:51  40.0  ...        10-50        Product
761  28/07/2020 04:03:13   NaN  ...        10-50        Product
762  28/07/2020 04:03:20   NaN  ...        10-50        Product
763  26/08/2020 09:06:44   NaN  ...          NaN            NaN
764  21/09/2020 01:47:10  31.0  ...        1000+        Product

[765 rows x 14 columns]


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import io
data_19= pd.read_csv(io.BytesIO(uploaded['Survey_2019.txt']))
print(data_19)

Saving Survey_2019.txt to Survey_2019 (1).txt
             Zeitstempel   Age  ... Company business sector   0
0    02.12.2019 11:18:26  33.0  ...                 Tourism NaN
1    02.12.2019 11:18:35  29.0  ...   Scientific Activities NaN
2    02.12.2019 11:18:56   NaN  ...                     NaN NaN
3    02.12.2019 11:19:08  30.0  ...               Transport NaN
4    02.12.2019 11:19:37  32.0  ...              Automotive NaN
..                   ...   ...  ...                     ...  ..
986  07.01.2020 09:23:01  30.0  ...                 Telecom NaN
987  07.01.2020 10:08:18  28.0  ...     Finance / Insurance NaN
988  07.01.2020 16:52:43  42.0  ...                Сommerce NaN
989  08.01.2020 11:18:41  33.0  ...                  Health NaN
990  08.01.2020 22:20:08  32.0  ...               Transport NaN

[991 rows x 23 columns]


In [ ]:
from google.colab import files
data_to_load = files.upload()

import pandas as pd
import io

data_20 = pd.read_csv(io.BytesIO(data_to_load['Survey_2020.txt']))
print(data_20)



Saving Survey_2020.txt to Survey_2020.txt
                Timestamp  ...  Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR
0     24/11/2020 11:14:15  ...                                                NaN                                                                      
1     24/11/2020 11:14:16  ...                                                NaN                                                                      
2     24/11/2020 11:14:21  ...                                                NaN                                                                      
3     24/11/2020 11:15:24  ...                                                NaN                                                                      
4     24/11/2020 11:15:46  ...                                                NaN                                                                      
...                   ...  ...                

# TASK-2

In [ ]:
from google.colab import files
uploaded = files.upload() 

import pandas as pd
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2018.txt']))
#Display the first 5 rows from 2018 survey data
result = df.head(5)
print("FIRST 5 ROWS OF THE 2018 SURVEY DATA")
print(result)

Saving Survey_2018.txt to Survey_2018 (7).txt
FIRST 5 ROWS OF THE 2018 SURVEY DATA
             Timestamp   Age  ... Company size   Company type
0  14/12/2018 12:41:33  43.0  ...     100-1000        Product
1  14/12/2018 12:42:09  33.0  ...       50-100        Product
2  14/12/2018 12:47:36  32.0  ...        1000+        Product
3  14/12/2018 12:50:15  25.0  ...        1000+        Product
4  14/12/2018 12:50:31  39.0  ...     100-1000  Ecom retailer

[5 rows x 14 columns]


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
#Dispaly  2020 survey data
df.info()

Saving Survey_2020.txt to Survey_2020 (3).txt
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1253 entries, 0 to 1252
Data columns (total 23 columns):
 #   Column                                                                                                                   Non-Null Count  Dtype  
---  ------                                                                                                                   --------------  -----  
 0   Timestamp                                                                                                                1253 non-null   object 
 1   Age                                                                                                                      1226 non-null   float64
 2   Gender                                                                                                                   1243 non-null   object 
 3   City                                                                                        

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
#Describe  2020 survey data
df.describe()


Saving Survey_2020.txt to Survey_2020 (5).txt


Age  ...  Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week
count  1226.000000  ...                                         373.000000                                                
mean     32.509788  ...                                          12.967828                                                
std       5.663804  ...                                          15.275174                                                
min      20.000000  ...                                           0.000000                                                
25%      29.000000  ...                                           0.000000                                                
50%      32.000000  ...                                           0.000000                                                
75%      35.000000  ...                                          30.000000                                                
max      69.000000  ...                                          40.000000                                                

[8 rows x 4 columns]

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
df.describe(include = np.object)



Saving Survey_2020.txt to Survey_2020 (6).txt


Timestamp  ... Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR
count                  1253  ...                                                462                                                                     
unique                 1248  ...                                                 59                                                                     
top     25/11/2020 18:28:01  ...                                                  0                                                                     
freq                      2  ...                                                161                                                                     

[4 rows x 19 columns]

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
df["City"]. value_counts()

Saving Survey_2020.txt to Survey_2020 (17).txt


Berlin             681
Munich             236
Frankfurt           44
Hamburg             40
Stuttgart           26
                  ... 
Friedrichshafen      1
Kiev                 1
Karlsruhe            1
NJ, USA              1
Bielefeld            1
Name: City, Length: 119, dtype: int64

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2018.txt']))
#Calculating ARITHMETIC MEAN
mean = df['Age'].mean()
print(mean)

Saving Survey_2018.txt to Survey_2018 (9).txt
32.183035714285715


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2018.txt']))
#Calculating MEDIAN
median = df['Age'].median()
print(median)

Saving Survey_2018.txt to Survey_2018 (10).txt
32.0


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2018.txt']))
#Calculating MODE
mode = df['Age'].mode()
print(mode)

Saving Survey_2018.txt to Survey_2018 (11).txt
0    30.0
dtype: float64


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2018.txt']))
#Checking MISSING VALUES in 2018 survey data
df.isnull()

Saving Survey_2018.txt to Survey_2018 (12).txt


Timestamp    Age  ...  Company size  Company type
0        False  False  ...         False         False
1        False  False  ...         False         False
2        False  False  ...         False         False
3        False  False  ...         False         False
4        False  False  ...         False         False
..         ...    ...  ...           ...           ...
760      False  False  ...         False         False
761      False   True  ...         False         False
762      False   True  ...         False         False
763      False   True  ...          True          True
764      False  False  ...         False         False

[765 rows x 14 columns]

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2018.txt']))
#COUNT OF THE MISSING VALUES FROM 2018 SURVEY DATA

print("show the details of  boolean data frame",df.isnull())
print(" Count total NaN in a DataFrame ", df.isnull().sum().sum()) 


Saving Survey_2018.txt to Survey_2018 (14).txt
show the details of  boolean data frame      Timestamp    Age  ...  Company size  Company type
0        False  False  ...         False         False
1        False  False  ...         False         False
2        False  False  ...         False         False
3        False  False  ...         False         False
4        False  False  ...         False         False
..         ...    ...  ...           ...           ...
760      False  False  ...         False         False
761      False   True  ...         False         False
762      False   True  ...         False         False
763      False   True  ...          True          True
764      False  False  ...         False         False

[765 rows x 14 columns]
 Count total NaN in a DataFrame  793


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2019.txt']))
#COUNT OF THE MISSING VALUES FROM 2019 SURVEY DATA

print("show the details of  boolean data frame",df.isnull())
print(" Count total NaN in a DataFrame ", df.isnull().sum().sum()) 

Saving Survey_2019.txt to Survey_2019 (3).txt
show the details of  boolean data frame      Zeitstempel    Age  ...  Company business sector     0
0          False  False  ...                    False  True
1          False  False  ...                    False  True
2          False   True  ...                     True  True
3          False  False  ...                    False  True
4          False  False  ...                    False  True
..           ...    ...  ...                      ...   ...
986        False  False  ...                    False  True
987        False  False  ...                    False  True
988        False  False  ...                    False  True
989        False  False  ...                    False  True
990        False  False  ...                    False  True

[991 rows x 23 columns]
 Count total NaN in a DataFrame  5724


In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
import numpy as np
import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
#COUNT OF THE MISSING VALUES FROM 2020 SURVEY DATA

print("show the details of  boolean data frame",df.isnull())
print(" Count total NaN in a DataFrame ", df.isnull().sum().sum()) 

Saving Survey_2020.txt to Survey_2020 (18).txt
show the details of  boolean data frame       Timestamp  ...  Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR
0         False  ...                                               True                                                                      
1         False  ...                                               True                                                                      
2         False  ...                                               True                                                                      
3         False  ...                                               True                                                                      
4         False  ...                                               True                                                                      
...         ...  ...                                         

# From 2018 survey data the number of missing values are 793, from 2019 there are 5724 values and from 2020 there are 3682 missing values. Over the years the missing values value is increasing and decreasing.

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd 
import numpy as np

import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
#Unique values from the column "Number of vacation days" of 2020 survey data
df['Number of vacation days'].unique()






Saving Survey_2020.txt to Survey_2020 (24).txt


array(['30', '28', '24', '29', '27', nan, '25', '31', '26', '60', '20',
       '22', '38', '35', '32', '40', '365', '36', '23', '33', '21',
       'unlimited', '14', 'unlimited ', '(no idea)',
       '30 in contract (but theoretically unlimited)', '0', 'Unlimited ',
       '15', '16', '3', '45', '~25', '12', '50', '23+', '99', 'Unlimited',
       '24 labour days', '37.5', '1', '5', '37', '39', '34', '10'],
      dtype=object)

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd 
import numpy as np

import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
#Frequency of the column "Number of vacation days" from 2020 survey data
len(df['Number of vacation days'].unique())




Saving Survey_2020.txt to Survey_2020 (25).txt


46

# TASK-3

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd 
import numpy as np

import io

df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
# Before renaming the columns
df.rename(columns = {'Position':'Position'}, inplace = True) 
# After renaming the columns 
print("\nAfter modifying first column:\n", df.columns)

Saving Survey_2020.txt to Survey_2020 (27).txt

After modifying first column:
 Index(['Timestamp', 'Age', 'Gender', 'City', 'Position ',
       'Total years of experience', 'Years of experience in Germany',
       'Seniority level', 'Your main technology / programming language',
       'Other technologies/programming languages you use often',
       'Yearly brutto salary (without bonus and stocks) in EUR',
       'Yearly bonus + stocks in EUR',
       'Annual brutto salary (without bonus and stocks) one year ago. Only answer if staying in the same country',
       'Annual bonus+stocks one year ago. Only answer if staying in same country',
       'Number of vacation days', 'Employment status', 'Сontract duration',
       'Main language at work', 'Company size', 'Company type',
       'Have you lost your job due to the coronavirus outbreak?',
       'Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week',
       'Have you received additional mo

In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd 
import numpy as np

import io

# checking whether 2020 survey data has any missing values or not
df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
df.isnull()


Saving Survey_2020.txt to Survey_2020 (28).txt


Timestamp  ...  Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR
0         False  ...                                               True                                                                      
1         False  ...                                               True                                                                      
2         False  ...                                               True                                                                      
3         False  ...                                               True                                                                      
4         False  ...                                               True                                                                      
...         ...  ...                                                ...                                                                      
1248      False  ...                                               True                                                                      
1249      False  ...                                              False                                                                      
1250      False  ...                                               True                                                                      
1251      False  ...                                              False                                                                      
1252      False  ...                                              False                                                                      

[1253 rows x 23 columns]

In [2]:
from google.colab import files
uploaded = files.upload()

import pandas as pd 
import numpy as np

import io
from sklearn.impute import SimpleImputer


df = pd.read_csv(io.BytesIO(uploaded['Survey_2020.txt']))
imputer_mean = SimpleImputer(missing_values=np.nan, strategy='mean')


Saving Survey_2020.txt to Survey_2020 (1).txt
